src/components/UserSelector/UserSelector.tsx

```diff
@@ -0,0 +1,120 @@
+import React, {
+  useState,
+  useRef,
+  useEffect,
+} from 'react';
+import classNames from 'classnames';
+import { getUsers } from '../../api/users';
+import { User } from '../../types/User';
+import { Error } from '../../types/Error';
+
+type Props = {
+  selectedUserId: number | null;
+  onSelectUser: (userId: number) => void;
+  onError: (error: Error | null) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  selectedUserId,
+  onSelectUser,
+  onError,
+}) => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<string | null>(null);
```

Comment:
The selectedUser state is not necessary. You can find the selected user's name directly from the users list using the selectedUserId.